core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -304,15 +308,18 @@ public void sendMessage(ReqT message) {
     Preconditions.checkState(stream != null, "Not started");
     Preconditions.checkState(!cancelCalled, "call was cancelled");
     Preconditions.checkState(!halfCloseCalled, "call was half-closed");
-    boolean failed = true;
+    Status failure = Status.CANCELLED;
     try {
       InputStream messageIs = method.streamRequest(message);
       stream.writeMessage(messageIs);
-      failed = false;
+      failure = null;
+    } catch (RuntimeException e) {
```

Comment:
Done.  

I vaguely remember statuses sometimes being lost in the case where there are multiple failures, but that might have been fixed.